pointillism-port.patch

```diff
@@ -0,0 +1,129 @@
+ from sugar3.graphics.toolbutton import ToolButton
+ from sugar3.activity.widgets import StopButton
+ from sugar3.activity.widgets import ActivityToolbarButton
+ from sugar3.graphics.toolbarbox import ToolbarBox
+-from sugar3.activity import activity
++# from sugar3.activity import activity
++import shutil
++import sys
++from sugarapp.widgets import SugarCompatibleActivity
++from sugarapp.widgets import DesktopSaveChooser
++from sugarapp.widgets import DesktopOpenChooser
++from sugar3.graphics.style import GRID_CELL_SIZE
++from sugarapp.helpers import PrimaryMonitor
+ from sugar3 import mime
+ import sugargame.canvas
+ import sugargame
+@@ -43,10 +50,10 @@ gi.require_version('Gdk', '3.0')
+ gi.require_version('Gtk', '3.0')
+ 
+ 
+-class Activity(activity.Activity):
++class Activity(SugarCompatibleActivity):
+ 
+     def __init__(self, handle):
+-        activity.Activity.__init__(self, handle)
++        SugarCompatibleActivity.__init__(self, handle)
+ 
+         self.max_participants = 1
+         self.file_path_temp = None
+@@ -56,13 +63,16 @@ class Activity(activity.Activity):
+         self.actividad = puntillism.Puntillism(self)
+ 
+         self.build_toolbar()
+-
++        
++        w = PrimaryMonitor.width()
++        h = PrimaryMonitor.height() - 1 * GRID_CELL_SIZE
++ 
+         self._pygamecanvas = \
+             sugargame.canvas.PygameCanvas(
+                 self,
+                 main=self.actividad.run,
+                 modules=[pygame.display])
+-
++        self._pygamecanvas.set_size_request(w, h)
+         self.set_canvas(self._pygamecanvas)
+         self._pygamecanvas.grab_focus()
+ 
+@@ -155,19 +165,18 @@ class Activity(activity.Activity):
+     def _savebutton_cb(self, button):
+         pygame.event.post(pygame.event.Event(
+             pygame.USEREVENT, action='savebutton'))
+-
++    
++    def can_close(self):
++        self.actividad.running = False
++        self.load_image_loop = False
++        pygame.quit()
++        return True
++    
+     def save_image(self, image):
+-        journalobj = datastore.create()
+-        journalobj.metadata['title'] = _('Pointillism')
+-        journalobj.metadata['mime_type'] = 'image/jpeg'
+-
+-        file_path = os.path.join(
+-            os.environ['SUGAR_ACTIVITY_ROOT'], 'data', 'pointillism.jpg')
+-
+-        pygame.image.save(image, file_path)
+-        journalobj.set_file_path(file_path)
+-        datastore.write(journalobj)
+-        journalobj.destroy()
++        chooser = DesktopSaveChooser(self, filename='pointiliism.png')
++        filename = chooser.get_filename()
++        if filename:
++            pygame.image.save(image, filename)
+ 
+     def open_image(self, button):
+         pygame.event.post(pygame.event.Event(
+@@ -176,26 +185,13 @@ class Activity(activity.Activity):
+     def choose_image_from_journal_cb(self):
+         ''' Create a chooser for image objects '''
+         self.image_id = None
+-        ##
+         self.chooser = None
```

Comment:
I meant, why doing `self.chooser = None` and then do `self.chooser = Desktop...` ? You can leave it out.

Reply:
Any further error handling will include self.chooser is not None  as its safer than a try..catch block. So its used here. Its better not to remove that for the sake of future updates. If you prefer to remove, it please tell me.